pkg/datapath/loader/base.go

```diff
@@ -170,7 +170,9 @@ func cleanIngressQdisc() error {
 
 // reinitializeIPSec is used to recompile and load encryption network programs.
 func (l *loader) reinitializeIPSec(ctx context.Context) error {
-	if !option.Config.EnableIPSec {
+	// If devices are specified, then we are relying on autodetection and don't
+	// need the code below, specific to EncryptInterface.
+	if !option.Config.EnableIPSec || len(option.Config.GetDevices()) > 0 {
```

Comment:
Just so I'm understanding correctly: if `--devices` are specified, we already attached `bpf_host` which already includes ipsec decryption logic so we need to skip attaching `bpf_network` because otherwise we potentially/most likely break other things right?

Reply:
Exactly. Goal is to deprecate `bpf_network` in a later release, then remove it completely. Using `bpf_host` is more consistent and needed to support IPsec+KPR.